src/Owin.Security.Providers.LinkedIn/LinkedInAuthenticationHandler.cs

```diff
@@ -271,5 +271,15 @@ protected override Task ApplyResponseChallengeAsync()
 
             return context.IsRequestCompleted;
         }
+
+        /// <summary>
+        ///     Gets proxy host name from <see cref="LinkedInAuthenticationOptions"/> if it is set.
+        ///     If proxy host name is not set, gets application request host name.
+        /// </summary>
+        /// <returns>Host name.</returns>
+        private string GetHostName()
+        {
+            return string.IsNullOrWhiteSpace(Options.ProxyHost) ? Request.Host.ToString() : Options.ProxyHost;
```

Comment:
don't proxies add the origin host as an `X-Forwarded-Host` header?

Reply:
Thanks for the suggestion :) I'll try that option with our proxy host.

Reply:
Unfortunately the X-Forwarded-Host header suggestion did not work for us. So we are using this solution in the meantime.

Reply:
Alrighty thanks @hokukim I'll merge this in tomorrow-ish

Reply:
Great, thank you!